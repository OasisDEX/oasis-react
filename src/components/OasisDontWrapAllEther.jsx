import React, { PureComponent } from 'react';
import { PropTypes } from 'prop-types';
// import ImmutablePropTypes from 'react-immutable-proptypes';

// import styles from './OasisDontWrapAllEther.scss';
import {InfoBox} from "./InfoBox";


const propTypes = PropTypes && {
};
const defaultProps = {};


class OasisDontWrapAllEther extends PureComponent {
  render() {
    return (
      <InfoBox color='danger' {...this.props}>
        <div style={{width: '100%'}}>
          <span style={{ textDecoration:'underline' }}>Do not wrap all</span> of your <b>ETH</b>! <br/>
          <div>
            Otherwise you will not be able to pay for transactions.
          </div>
        </div>
      </InfoBox>
    );
  }
}

OasisDontWrapAllEther.displayName = 'OasisDontWrapAllEther';
OasisDontWrapAllEther.propTypes = propTypes;
OasisDontWrapAllEther.defaultProps = defaultProps;
export default OasisDontWrapAllEther;
