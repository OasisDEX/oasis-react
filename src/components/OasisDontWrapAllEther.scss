.warningIco {
  margin-left: 0.25em;
  margin-top: -2px;
}

.tokenName {
}
.text {
  display: inline-block;
  margin-left: 5px;
}
