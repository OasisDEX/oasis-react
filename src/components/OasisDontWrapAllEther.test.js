/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';
import { shallow } from 'enzyme';
import OasisDontWrapAllEther from './OasisDontWrapAllEther';


describe('(Component) OasisDontWrapAllEther', () => {
  it('should render', () => {
    const props = {
      children: (
        <span>test</span>
      )
    };
    const wrapper = shallow(
      <OasisDontWrapAllEther {...props}/>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
