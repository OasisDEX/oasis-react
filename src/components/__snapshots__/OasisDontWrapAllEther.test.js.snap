// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) OasisDontWrapAllEther should render 1`] = `
<InfoBox
  className=""
  color="danger"
  size="md"
>
  <div
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <span
      style={
        Object {
          "textDecoration": "underline",
        }
      }
    >
      Do not wrap all
    </span>
     of your 
    <b>
      ETH
    </b>
    ! 
    <br />
    <div>
      Otherwise you will not be able to pay for transactions.
    </div>
  </div>
</InfoBox>
`;
